polaris-react/src/components/Button/Button.tsx

```diff
@@ -41,8 +41,10 @@ export interface ButtonProps extends BaseButton {
   dataPrimaryLink?: boolean;
   /** Sets the color treatment of the Button. */
   tone?: 'critical' | 'success';
-  /** Changes the visual appearance of the Button. */
-  variant?: 'plain' | 'primary' | 'tertiary' | 'monochromePlain';
+  /** Changes the visual appearance of the Button.
+   * @default 'secondary'
+   */
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'plain' | 'monochromePlain';
```

Comment:
I don't think adding `secondary` is necessary. If we add it, then `variant` should be required and it will require re-running migrations for all default buttons to add `secondary`. If the variant is not required, then default button can be rendered two ways, either with no variant prop or with `variant='secondary'` which is also confusing.

Reply:
It's not doing anything right now, so not necessary. I added it because it's self documenting that secondary is the default button style. Found it weird that only primary and tertiary were mentioned. 

I actually forgot that this was part of this PR though so I'll remove it. With Chloe's changes to button(s)From this PR might not be needed so I'm going to leave it and work on other stuff for now